DataFormats/Headers/include/Headers/DataHeader.h

```diff
@@ -25,6 +25,7 @@
 #include <iostream>
 #include <memory>
 #include <cassert>
+#include <cstring>
```

Comment:
this is needed for `memcmp`, should probably be added as a comment. I somehow feel that it's strange to include the `cstring` header in this general data format header. But the `memcmp` implementation obviously does a good job here.